http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1478,41 +756,181 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
           AppControlledResourceStatus.Created
         )
         .transaction
-    } yield dbName
+    } yield result
   }
 
-  private[util] def deleteAppWsmResources(app: App, workspaceId: WorkspaceId)(implicit
+  private[util] def createWsmKubernetesNamespaceResource(app: App,
+                                                         workspaceId: WorkspaceId,
+                                                         namespacePrefix: String,
+                                                         databases: List[UUID],
+                                                         identity: Option[UUID]
+  )(implicit ev: Ask[F, AppContext]): F[CreatedControlledAzureKubernetesNamespaceResult] =
+    for {
+      ctx <- ev.ask
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Creating $namespacePrefix namespace for app ${app.appName.value} in cloud workspace ${workspaceId.value}"
+      )
+
+      // Build WSM clients
+      wsmApi <- buildWsmControlledResourceApiClient
+      wsmResourceApi <- buildWsmResourceApiClient
+
+      // Name of the WSM resource. Must be unique per workspace.
+      // For shared apps, name it by the appType so it's semantically meaningful.
+      // There can only be at most 1 shared app type per workspace anyway.
+      // For private apps, use the namespacePrefix to ensure uniqueness.
+      wsmResourceName = app.samResourceId.resourceType match {
+        case SamResourceType.SharedApp => s"${app.appType.toString.toLowerCase}-ns"
+        case _                         => namespacePrefix
+      }
+
+      // Obtain external database IDs by querying WSM and filtering by name
+      // TODO this will get simpler once WSM createNamespace supports passing databases by name instead of UUID
```

Comment:
Is there an existing ticket for this work that could be linked here so we know when it's not a blocker anymore?

Reply:
I'll piggy back on https://broadworkbench.atlassian.net/browse/WOR-1250, will update this comment.